### PR TITLE
Fixed importer issue

### PR DESCRIPTION
**Pull request by nazrulcse**
Fixed Character encoding issue and Find out invalid email.

### CODE REVIEW

File: app/controllers/lists_controller.rb [hunk 1]
```diff
@@ -65,11 +67,20 @@ def destroy
...
+    #ImportListFile.perform_later(@list, params[:file].read)
     #@list.import_members(params[:file])
     # @list.save
 
-    redirect_to @list, notice: "List will be imported shortly. Please refresh the page in a moment."
+    if invalid_rows.present?
+      session[:invalid_rows] = invalid_rows
+    end
+
+      redirect_to @list, notice: "List will be imported shortly. Please refresh the page in a moment"
```
Comment (codeguardci) [inline app/controllers/lists_controller.rb]: Line is too long. [101/100]<br>Inconsistent indentation detected.

File: app/controllers/lists_controller.rb [hunk 2]
```diff
@@ -109,26 +120,26 @@ def clear
 
   private
 
-  def set_list
-    @list = current_user.lists.find(params[:id])
-  end
+    def set_list
```
Comment (codeguardci) [inline app/controllers/lists_controller.rb]: Inconsistent indentation detected.

File: app/controllers/lists_controller.rb [hunk 3]
```diff
@@ -109,26 +120,26 @@ def clear
...
+      @list = current_user.lists.find(params[:id])
+    end
 
-  def safe_params
-    params.require(:list).permit(:name, :default_sender_name, :default_sender_email)
-  end
+    def safe_params
```
Comment (codeguardci) [inline app/controllers/lists_controller.rb]: Inconsistent indentation detected.

File: app/controllers/lists_controller.rb [hunk 4]
```diff
@@ -109,26 +120,26 @@ def clear
...
+      params.require(:list).permit(:name, :default_sender_name, :default_sender_email)
+    end
 
-  def prepare_liquid_tags_params
-    query = []
-    params[:segment][:query_params].values.each do |el|
-      if el.fetch("_destroy") == ""
-        el.delete("_destroy")
-        query << el
+    def prepare_liquid_tags_params
```
Comment (codeguardci) [inline app/controllers/lists_controller.rb]: Inconsistent indentation detected.

File: app/controllers/lists_controller.rb [hunk 5]
```diff
@@ -109,26 +120,26 @@ def clear
...
       end
+      query
     end
-    query
-  end
 
-  def add_abilities
-    abilities << List
-  end
+    def add_abilities
```
Comment (codeguardci) [inline app/controllers/lists_controller.rb]: Inconsistent indentation detected.

File: app/controllers/lists_controller.rb [hunk 6]
```diff
@@ -65,11 +67,20 @@ def destroy
   end
 
   def import
-    ImportListFile.perform_later(@list, params[:file].read)
+    # ImportListFile.perform_later(@list, params[:file].read)
+    @not_imported = ""
+    invalid_rows = @list.import_members(params[:file])
+    @list.save
+    #ImportListFile.perform_later(@list, params[:file].read)
```
Comment (codeguardci) [inline app/controllers/lists_controller.rb]: Missing space after #.

File: app/models/list.rb [hunk 7]
```diff
@@ -26,20 +26,21 @@ class List < ActiveRecord::Base
   def import_members(file)
     result = import(file)
     current_list = list_members || []
+   # current_list += result[:valid_rows]
```
Comment (codeguardci) [inline app/models/list.rb]: Incorrect indentation detected (column 3 instead of 4).

File: app/services/list_importer.rb [hunk 8]
```diff
@@ -74,7 +81,7 @@ def open_file(file)
           csv_file = file
         else
           uploaded_file.rewind
-          csv = uploaded_file.read.encode("UTF-8", invalid: :replace).gsub!(/\r\n?/, "\n")
+          csv = uploaded_file.read.encode("UTF-8","ISO-8859-2", invalid: :replace).gsub!(/\r\n?/, "\n")
```
Comment (codeguardci) [inline app/services/list_importer.rb]: Line is too long. [103/100]<br>Space missing after comma.

File: app/services/list_importer.rb [hunk 9]
```diff
@@ -51,7 +56,9 @@ def process(header, parsed_file, first_data_row)
         next
       end
     end
-    rows
+    process_result[:valid_rows] = rows
+    process_result[:invalid_rows] = invalid_rows
+    return process_result
```
Comment (codeguardci) [inline app/services/list_importer.rb]: Redundant `return` detected.

File: app/services/list_importer.rb [hunk 10]
```diff
@@ -1,17 +1,18 @@
 module ListImporter
   require "roo"
-  COLUMN_SEPARATORS = [",", ";", "\t", ":", "|", " "]
+  COLUMN_SEPARATORS = [",", ";", "\t", ":", "|"," "]
```
Comment (codeguardci) [inline app/services/list_importer.rb]: Space missing after comma.

File: app/services/list_importer.rb [hunk 11]
```diff
@@ -94,7 +101,8 @@ def open_file(file)
   end
 
   def find_col_sep(file)
-    first_line = File.open(file.tempfile).first.encode("UTF-8", invalid: :replace)
+    first_line = File.open(file.tempfile).first.encode("UTF-8","ISO-8859-2", invalid: :replace)
```
Comment (codeguardci) [inline app/services/list_importer.rb]: Space missing after comma.

File: app/models/list.rb [hunk 12]
```diff
@@ -26,20 +26,21 @@ class List < ActiveRecord::Base
   def import_members(file)
     result = import(file)
     current_list = list_members || []
+    list = result[:valid_rows] || []
+   # current_list += result[:valid_rows]
```
Comment (codeguardci) [inline app/models/list.rb]: Incorrect indentation detected (column 3 instead of 4).

File: app/services/list_importer.rb (hunk 9 above)
Comment (olimart) [inline app/services/list_importer.rb]: @nazrulcse please remove redundancy.

File: app/controllers/lists_controller.rb (hunk 6 above)
Comment (olimart) [inline app/controllers/lists_controller.rb]: @nazrulcse please fix as well

File: app/services/list_importer.rb (hunk 8 above)
Comment (olimart) [inline app/services/list_importer.rb]: @nazrulcse please fix as well

File: app/services/list_importer.rb (hunk 11 above)
Comment (olimart) [inline app/services/list_importer.rb]: @nazrulcse please fix as well

**Pull request merged**